charts/haproxy/templates/service.yaml

```diff
@@ -6,6 +6,7 @@ metadata:
     {{- include "haproxy.labels" . | nindent 4 }}
 spec:
   type: {{ .Values.service.type }}
+  clusterIP: {{ .Values.service.clusterIP | quote}}
```

Comment:
I think `clusterIP` should be set only if is not empty, otherwise updating the service multiple times may lead to errors, in the older k8s versions.
Please see https://github.com/helm/charts/pull/8221